.github/workflows/build_and_deploy_ext2_img.yml

```diff
@@ -0,0 +1,133 @@
+  
+  # Lets you supply arguments when you run a github action manually.
+    inputs:
+      DOCKER_BUILD_CONTEXT:
+        type: string
+        description: 'Path or url to the docker build context.'
+        required: true
+        default: 'dockerfiles/example'
+  
+      IMAGE_SIZE:
+        type: string
+        description: 'Size of the base ext2 image. 1GB is the max size for pages, 2GB for Gh release.. Refer to the fallocate(1) manpage for multiplicative suffixes'
+        required: true
+        default: '950MB'
+
+      IMAGE_NAME:
+        type: string
+        description: 'Name of final image. Example final url: https://www.github.io/<owner>/<repositoryName>/<image_name>'
+        required: true
+        default: 'base_image.ext2'
+
+      DEPLOY_TO_PAGES:
+        type: boolean
+        description: 'Deploy to pages.'
+        required: true
+        default: true
+
+      GITHUB_RELEASE:
+        type: boolean
+        description: 'Upload github release.'
+        required: true
+        default: true
+
+      GITHUB_RELEASE_TAG:
+        type: string
+        description: 'Tag for the github release. Optional. Defaults to ext2_image'
+        required: false
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest # Image to run the worker on.
+    
+    # Our variables
+    env:
+      DOCKER_BUILD_CONTEXT: '${{ github.event.inputs.DOCKER_BUILD_CONTEXT }}'
+      TAG: "ext2-webvm-base-image"
+      IMAGE_SIZE: '${{ github.event.inputs.IMAGE_SIZE }}'
+      IMAGE_NAME: '${{ github.event.inputs.IMAGE_NAME }}'
+      DEST: /assets/images/tmp/ # Path to directory where we host the final image from.
+
+    permissions:
+      contents: write    # Required permission to make a github release.  
+  
+    steps:
+      # Checks-out our repository under $GITHUB_WORKSPACE, so our job can access it
+      - uses: actions/checkout@v3
+      # We make the directory to host our image from.
+      - run: sudo mkdir -p $DEST
+      # We build the i386 Dockerfile image.
+      - run: docker build $DOCKER_BUILD_CONTEXT --tag $TAG --platform=i386
+      # We run the image so that we can export the container.
+      - run: docker run -d $TAG
+      # We create our ext2 filesystem.
+      # We preallocate space for the ext2 image
+      - run: sudo fallocate -l $IMAGE_SIZE ${DEST}${IMAGE_NAME}
+      # Format to ext2 linux kernel revision 0
+      - run: sudo mkfs.ext2 -r 0 ${DEST}${IMAGE_NAME}
+      # We mount the ext2 image in order to modify it.
+      - run: | 
+          sudo mkdir -p /mnt/ext2_image
+          sudo mount -o loop -t ext2 ${DEST}/${IMAGE_NAME} /mnt/ext2_image
+      # We export our container its filesystem and unpack it into the mounted ext2 image.
+      - run: | 
+          docker export $(sudo docker ps -aq) | sudo tar -x -C /mnt/ext2_image
+          sudo umount /mnt/ext2_image
+  
+      # Result is an ext2 image for webvm.
+      - name: Upload GitHub Pages artifact
+        uses: actions/upload-pages-artifact@v1.0.7
```

Comment:
We dont' even need the artifacts at this point. Pages + releases are enough

Reply:
To my understanding we need to upload the Github pages artifact, in order to deploy it to pages with the Gh deploy pages action. 

https://github.com/actions/deploy-pages/tree/v2.0.0/

Reply:
Ok, you can anyway reduce the days to the default. The actual output for our own deployment will be a release.